DSpace/dspace-angular
# Private/Withdrawn badges for items everywhere

## References
This PR fixes #947

## Description
This PR refactors the badges shown for withdrawn and private items on the administrative search to be shown on every listable component so that they're visible on other places as well (e.g. regular search, collection pages...)
It also adds an alert on top of simple and full item pages mentioning whether or not the item is private or withdrawn.

## Instructions for Reviewers
Changes made:
- Moved badges template to `ListableObjectComponentLoaderComponent`.
-- This component will determine whether or not the badges should be displayed (ViewMode is set to a list and the object contains `isDiscoverable` and/or `isWithdrawn` properties)
-- The badges template is passed on to the loaded list component, which knows how to deal with the content (e.g. a grid element will place the badge in the top-left corner of the thumbnail)
- Added `ItemAlertsComponent`, displaying warning alerts for private and withdrawn items and rendered on top of simple and full item pages
- Added/Modified test cases reflecting these changes

How to test:
- Anonymous users should never see these badges or alerts, as they don't have the rights to view these items
- Log in as an administrator and visit administrative search, regular search and collection pages and verify the badges are visible for private and withdrawn items. Also verify their position is on top of the element for lists and top-left within the thumbnails for grids.
- Log in as an administrator and visit some of these item pages (simple & full) and verify an alert is displayed on top of the page mentioning the item being private or withdrawn.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
